app/core/store/modules/classrooms.js

```diff
@@ -0,0 +1,103 @@
+import classroomsApi from 'core/api/classrooms'
+
+export default {
+  namespaced: true,
+
+  state: {
+    loading: {
+      byClassroom: {},
+      byTeacher: {}
+    },
+
+    classrooms: {
+      byClassroom: {},
+      // Classrooms by teacher ID
+      //  {
+      //     active: [],
+      //     archived: []
+      //  }
+      byTeacher: {}
+    }
+  },
+
+  mutations: {
+    toggleLoadingForTeacher: (state, teacherId) => {
+      let loading = true
+      if (state.loading.byTeacher[teacherId]) {
+        loading = false
+      }
```

Comment:
Can this be `const loading = !state.loading.byTeacher[teacherId]`?

Reply:
Yep - I try to be verbose with this kind of logic because it increases readability over time but this one is so straightforward I don't think there's a lot of value in it, updated.